mux.go

```diff
@@ -0,0 +1,48 @@
+package sm_yamux
+
+import (
+	"io/ioutil"
+	"net"
+
+	mux "github.com/libp2p/go-libp2p-core/mux"
+	yamux "github.com/libp2p/go-yamux"
+)
+
+var DefaultTransport *Multiplexer
+
+func init() {
+	config := yamux.DefaultConfig()
+	// We've bumped this to 16MiB as this critically limits throughput.
+	//
+	// 1MiB means a best case of 10MiB/s (83.89Mbps) on a connection with
+	// 100ms latency. The default gave us 2.4MiB *best case* which was
+	// totally unacceptable.
+	config.MaxStreamWindowSize = uint32(16 * 1024 * 1024)
+	// don't spam
+	config.LogOutput = ioutil.Discard
+	// We always run over a security transport that buffers internally
+	// (i.e., uses a block cipher).
+	config.ReadBufSize = 0
+	DefaultTransport = (*Multiplexer)(config)
+}
+
+// Multiplexer implements mux.Multiplexer that constructs
+// yamux-backed muxed connections.
+type Multiplexer yamux.Config
```

Comment:
It's still a transport (a multiplexing transport). We don't need to rename this so let's avoid introducing unnecessary breaking changes.

Reply:
Note: I agree in principle; we're overloading "transport" everywhere. It's just not worth the hassle to change it here (especially because we can't be consistent without introducing a major breaking change (https://github.com/libp2p/go-libp2p-yamux/pull/10/files#r385778045)).

Reply:
Ok, then will revert that